src/pages/signup/__tests__/signup.component.test.js

```diff
@@ -0,0 +1,89 @@
+import { screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import wrapper from 'test/test-utils';
+
+import SignUp from '../signup.component';
+
+let mockIsLoading = false;
+
+jest.mock('service/signup', () => ({
+  useMutationSignup: () => ({
+    isLoading: mockIsLoading,
+  }),
+}));
+
+describe('<SignUp />', () => {
+  it('renders the title and all the inputs', () => {
+    wrapper(<SignUp />);
+    expect(
+      screen.getByRole('heading', { name: /cadastre-se/i })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('textbox', { name: /nome completo/i })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('textbox', { name: /e-mail/i })
+    ).toBeInTheDocument();
+    expect(screen.getByRole('textbox', { name: /cpf/i })).toBeInTheDocument();
+    expect(
+      screen.getByRole('textbox', { name: /telefone/i })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('textbox', { name: /endereço/i })
+    ).toBeInTheDocument();
+    expect(screen.getByLabelText(/^senha/i)).toBeInTheDocument();
+    expect(screen.getByLabelText(/confirmar senha/i)).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: /cadastrar/i })
+    ).toBeInTheDocument();
+  });
+
+  it('first renders with a disabled submit button', () => {
+    wrapper(<SignUp />);
+    expect(screen.getByRole('button', { name: /cadastrar/i })).toBeDisabled();
+  });
+
+  it('should enable submit button when validation passes', async () => {
```

Comment:
Since there is a test to ensure the submit button is disabled by default and this is the **only** test that tests the validation of the form I think we should have some concerns about it:

1. Maybe the component enables the button after the user fills out the first input field;
2. Maybe the component enables the button after the user fills out any input field;
3. Maybe the component enables the button after the user fills out the last input field;
4. Maybe the component enables the button after the user fills out all input fields but it doesn't care about the validation of each input field, it only cares if all the inputs are not empty.